ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Pipelines.resource

```diff
@@ -112,6 +112,24 @@ Click Action From Actions Menu
     Wait Until Page Contains Element    //table//tr[td[*[a[text()='${pipeline_name}']]]]//td[contains(@class, 'pf-c-table__action')]/div/ul/li/button[contains(text(), '${action}')]
     Click Element    //table//tr[td[*[a[text()='${pipeline_name}']]]]//td[contains(@class, 'pf-c-table__action')]/div/ul/li/button[contains(text(), '${action}')]
 
+Pipeline Context Menu Should Be Readonly
+    [Documentation]   Test Pipeline YAML context menu and verify read only (https://github.com/opendatahub-io/odh-dashboard/issues/1689)
+    [Arguments]    ${pipeline_name}
+    ${orig wait}=    SeleniumLibrary.Set Selenium Implicit Wait    10s
+    Wait Until Element is Visible     //button[contains(text(), "Data Science Pipelines")]          timeout=30
+    Click Button      Data Science Pipelines
+    Click Link      Pipelines
+    Wait Until Element is Visible     //a[text()="${pipeline_name}"]          timeout=30s
+    Click Link      ${pipeline_name}
+    Click Element     //button[@aria-label="Pipeline YAML Tab"]
+    Open Context Menu    //div[contains(@class, 'lines-content')]
+    Click Element     //button[@aria-label="Pipeline YAML Tab"]
```

Comment:
by looking at the GH issue, i understood that there are 2 different scenario:
1. the pipeline YAML editor should be read only
2. the Command Palette should not allow to run commands. I guess this is the "Context Menu" this keyword is referring.

Is my understanding correct? If yes, I would expect this "Open Context Menu" to be placed either in a separated keyword or after checking the Pipeline YAML editor is read-only

Reply:
The commands menu should allow to run commands, and this is tested just by a single line of code:
`Press Keys    //div[contains(@class, 'lines-content')]    F1+fold+SPACE+all+ENTER`

If I move this line to a new keyword, we will have to rerun the prerequisites to reach a pipeline yaml in that kw, which will also take more execution time, and duplicating code.

In case we decide to add a complete new test, for example to test all the available command options - if we see it is high priority use case (I don't think it is though) - we can consider to add a new test with more keywords.

For current use case I believe keeping the readonly validation + context check in same keyword is fine.

Reply:
My concern is that we don't check that the YAML is read-only by trying editing it without using the context-menu.

For example, you could have the pre-requisites in the test. Then you have a kw to test the context menu and you have a kw to try editing the YAML from the Editor (without context menu).

Reply:
In the context menu the "edit" commands do not exists. That's why I need to send keyboard types to check if edit is not allowed.

Reply:
yeah I'm suggesting to try editing without leveraging on context menu

Reply:
Yes, that is what the logic does: 
1) Open Context Menu with mouse to verify it is available.
2) Close the menu by clicking the YAML again
3) Open the context menu with F11 and send command to fold all.
4) Try to edit the YAML without the context menu, by sending keyboard types

Reply:
could you make more clear it in the code? The kw is called "Pipeline Context Menu Should Be Readonly" but actually mixes the context menu with Editor check. I think also for future maintanance it would be better to have clear steps in the test.
You could move the pre-requisites steps outside the kw and run them once. But then with separated keywords you check the editor is read-only in any available scenarios (context menu vs direct typing in the Editor)

Reply:
Separated to 2 keywords:

![image](https://github.com/red-hat-data-services/ods-ci/assets/9913945/d4c3a366-20cf-4f80-9630-962ea254b2c0)

Reply:
nope, it's testing one of the available commands ("fold all"), and verifies both mouse reaction and keyboard